### ISSUE TITLE
Scan  Not detecting RMi

### ISSUE THREAD

**Issue by random-robbie** [qtc-de/remote-method-guesser#62]
Hey,

I was testing this tool out but everytime i chose a rmi from shodan on port 1099 scan never detected it.

```
java -jar rmg-5.0.0-jar-with-dependencies.jar scan 195.187.71.22
[+] Scanning 112 Ports on xxx.xxx.xxx.xx for RMI services.
[+]
[+] 	[224 / 224] [#############################] 100%
[+]
[+] Portscan finished.
(base) $ nc xxx.xxx.xxxx 1099
��srjava.rmi.MarshalledObject|���c�>IhashlocBytest[BobjBytesq~xp�Yu
                                                                   ur[B��T�xp+��thttp://xx.xx.xx.xx:8083/q~q~uq~Ĭ�sr org.jnp.server.NamingServer_Stubxrjava.rmi.server.RemoteStub���ɋ�exrjava.rmi.server.RemoteObject�a��
                                                                                                                                                                                                                           a3xpw8
�~���񊬿�x
```

**Comment by qtc-de** [comment]
Hi @random-robbie :wave: 

the scanner was never intended to be reliable, but you are totally correct, in the current implementation it almost fails every time. I think this was not always the case, but maybe it only worked in my test setup with basically no latency on the connection :thinking: I have to dig into this and check how we can make it more reliable :+1:

**Comment by qtc-de** [comment]
Hi @random-robbie :wave: 

I finally found time to look at this issue and it seems that I managed to solve it. The solution was simple:

During early development of this feature, I hard coded the socket read and connect timeouts into the `PortScanner` Class with values `5` and `5`. The error here was that I expected these timeouts to be in seconds, but they have to be specified in milliseconds. Since I was only testing locally, 5 milliseconds still worked fine.

Later when adding command line options for this action, I chose defaults of `5000` for read and `3000` for connect. However, I never applied these options and the old defaults of `5` and `5` were still in use.

I applied the fixes to the development branch. It was again only tested locally so far, but I still expect that the `scan` action is now way more reliable. I will leave this issue open until the changes were merged to master.

Thanks again for reporting :+1:
